sample-app/src/components/InputDropdown.tsx

```diff
@@ -70,11 +70,12 @@ export default function InputDropdown({
 }: React.PropsWithChildren<Props>): JSX.Element | null {
   const [{
     focusedSectionIndex,
-    shouldDisplayDropdown,
+    shouldDisplayDropdown: shouldDisplayDropdownWithoutForcedHide,
   }, dispatch] = useReducer(reducer, {
     focusedSectionIndex: undefined,
     shouldDisplayDropdown: false,
   });
+  const shouldDisplayDropdown = shouldDisplayDropdownWithoutForcedHide && !hideDropdown;
```

Comment:
I would argue that the true `shouldDisplayDropdown` is the one defined here. Perhaps we could rename the `shouldDisplayDropdown` in the reducer to something else such as `dropdownHidden`. That would let us avoid the rename to `shouldDisplayDropdownWithoutForcedHide`. It might read a little strange to have  `const shouldDisplayDropdown = !dropdownHidden && !hideDropdown;`, however we could add comment explaining that we should only show the dropdown if both the state and props allow it

Reply:
renamed shouldDisplayDropdown in the reducer to dropdownHidden and the hideDropdown prop to forceHideDropdown. it's still a code smell but we do have to plan the next (and hopefully last) refactor of InputDropdown after v1 of the sample app